Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -2389,8 +2393,19 @@ bool LocalFrameView::scrollToFragmentInternal(StringView fragmentIdentifier)
         scrollPositionAnchor = m_frame->document();
     maintainScrollPositionAtAnchor(scrollPositionAnchor.get());
     
-    // If the anchor accepts keyboard focus, move focus there to aid users relying on keyboard navigation.
     if (anchorElement) {
+        if (m_frame->settings().detailsAutoExpandEnabled()) {
+            for (auto& element : ancestorsOfType<HTMLElement>(*anchorElement)) {
+                if (is<HTMLSummaryElement>(element))
+                    continue;
+                if (is<HTMLDetailsElement>(element) && !element.hasAttributeWithoutSynchronization(HTMLNames::openAttr)) {
+                    ShouldNotFireMutationEventsScope scope(document);
+                    downcast<HTMLDetailsElement>(element).toggleOpen();
+                    break;
+                }
+            }
+        }
```

Comment:
This code is duplicated with FindController.cpp, can we make a common method on Element or such ?

Reply:
> This code is duplicated with FindController.cpp, can we make a common method on Element or such ?

Yes — made it so (though made it on Node rather than on Element, because the code where the caller is in FindController.cpp operates on a Node, not an element)